src/QirRuntime/test/QIR-static/qir-test-ouput.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include <iostream>
+#include <sstream>
+
+#include "catch.hpp"
+
+#include "qirTypes.hpp"
+#include "quantum__qis_internal.hpp"
+
+extern "C" void Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(QirString*);        // NOLINT
+
+
+// https://stackoverflow.com/a/5419388/6362941
+// https://github.com/microsoft/qsharp-runtime/pull/511#discussion_r574170031
+// https://github.com/microsoft/qsharp-runtime/pull/511#discussion_r574194191
+struct QOstreamRedirector
```

Comment:
Nit: personally, I like to indicate in the name of such types that they are meant to be scoped (e.g. `QirContextScope`), because then it's easier to see why you might want a block of {} in the code that uses them.

Reply:
Sounds very reasonable to me...
Renaming to `OstreamRedirectorScoped`.